File: scratch/test_validation.py
import requests

yaml_content_1 = """---
title: Mediflow Proactive Monitor
emoji: 🛡️
colorFrom: green
colorTo: blue
sdk: gradio
sdk_version: 4.44.1
python_version: "3.10"
app_file: app.py
pinned: false
---"""

yaml_content_2 = """---
title: Mediflow Proactive Monitor
emoji: 🛡️
colorFrom: green
colorTo: blue
sdk: gradio
sdk_version: 4.44.1
python_version: 3.10
app_file: app.py
pinned: false
---"""

def test_validate(content):
    print("Testing content:")
    print(content)
    try:
        url = "https://huggingface.co/api/validate-yaml"
        res = requests.post(url, data=content.encode("utf-8"), headers={"Content-Type": "text/plain"})
        print(f"Status Code: {res.status_code}")
        print(f"Content-type: {res.headers.get('Content-Type')}")
        print(f"Response Content: {res.text[:500]}")
    except Exception as e:
        print(f"Exception: {e}")
    print("-" * 40)

print("Test 1 (quoted 3.10):")
test_validate(yaml_content_1)

print("Test 2 (unquoted 3.10):")
test_validate(yaml_content_2)

File: scratch/test_validation.js
import https from 'https';

const yaml_content_1 = `---
title: Mediflow Proactive Monitor
emoji: 🛡️
colorFrom: green
colorTo: blue
sdk: gradio
sdk_version: 4.44.1
python_version: 3.11
app_file: app.py
pinned: false
---`;

const yaml_content_2 = `---
title: Mediflow Proactive Monitor
emoji: 🛡️
colorFrom: green
colorTo: blue
sdk: gradio
sdk_version: 4.44.1
python_version: 3.12
app_file: app.py
pinned: false
---`;

const yaml_content_3 = `---
title: Mediflow Proactive Monitor
emoji: 🛡️
colorFrom: green
colorTo: blue
sdk: gradio
sdk_version: 4.44.1
app_file: app.py
pinned: false
---`;

function testValidate(content, name) {
  return new Promise((resolve) => {
    console.log(`\nTesting ${name}:`);
    const req = https.request({
      hostname: 'huggingface.co',
      path: '/api/validate-yaml',
      method: 'POST',
      headers: {
        'Content-Type': 'text/plain',
        'Content-Length': Buffer.byteLength(content)
      }
    }, (res) => {
      let data = '';
      res.on('data', (chunk) => { data += chunk; });
      res.on('end', () => {
        console.log(`Status Code: ${res.statusCode}`);
        console.log(`Content-Type: ${res.headers['content-type']}`);
        console.log(`Response: ${data.slice(0, 500)}`);
        resolve();
      });
    });
    req.on('error', (e) => {
      console.error(`Error: ${e.message}`);
      resolve();
    });
    req.write(content);
    req.end();
  });
}

async function main() {
  await testValidate(yaml_content_1, 'python_version: 3.11');
  await testValidate(yaml_content_2, 'python_version: 3.12');
  await testValidate(yaml_content_3, 'Without python_version');
}

main();
